examples/aes-example.cpp

```diff
@@ -0,0 +1,167 @@
+
+static const std::vector<uint8_t> plaintext = utility::fromHex("deadbeef");
+static const SymmetricCipherMode aeOperationMode = SymmetricCipherMode::GCM;
+static const SymmetricCipherMode plainOperationMode = SymmetricCipherMode::CBC;
+static const SymmetricCipherPadding padding = SymmetricCipherPadding::PKCS;
+static const std::vector<uint8_t> aesKey = utility::fromHex("000102030405060708090a0b0c0d0e0f");
+static const std::vector<uint8_t> authData = utility::fromHex("beefdead");
+static const std::vector<uint8_t> iv = utility::fromHex("00102030405060708090a0b0c0d0e0f0");
+static const size_t authTagLength = 16;
+
+struct AesAuthenticatedEncryptionResult {
+    std::vector<uint8_t> iv;
+    std::vector<uint8_t> ciphertext;
+    std::vector<uint8_t> authTag;
+};
+
+struct AesEncryptionResult {
+    std::vector<uint8_t> iv;
+    std::vector<uint8_t> ciphertext;
+};
+
+struct AesAuthenticatedEncryptionResult aesAuthenticatedEncryption()
+{
+    auto encryptorBuilder = AESCipherBuilder{aeOperationMode, SymmetricCipherKeySize::S_128, aesKey}
+            // Padding mode is optional. Default is SymmetricCipherPadding::PKCS
+            .setPadding(padding)
+            // The IV is optional. Default is a random value
+            .setIV(iv);
+    // the authTagLength is an optional value
+    encryptorBuilder.setAuthTagLength(authTagLength);
+
+    auto authenticatedEncryptor = encryptorBuilder.buildAuthenticatedEncryptor();
+    authenticatedEncryptor->addAssociatedData(authData);
+
+    std::vector<uint8_t> ciphertext;
+    try {
+        authenticatedEncryptor->update(plaintext);
+        ciphertext = authenticatedEncryptor->finish();
+    }  catch (openssl::OpenSSLException &e) {
+        std::cerr << "Failure encrypting the data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (MoCOCrWException &e) {
+        std::cerr << "Failure encrypting the data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    return AesAuthenticatedEncryptionResult{
+        authenticatedEncryptor->getIV(),
+        ciphertext,
+        authenticatedEncryptor->getAuthTag()
+    };
+}
+
+void aesAuthenticatedDecryption(struct AesAuthenticatedEncryptionResult aesEncryptResult)
+{
+    auto decryptorBuilder = AESCipherBuilder{aeOperationMode, SymmetricCipherKeySize::S_128, aesKey}
+            .setPadding(padding)
+            .setIV(iv);
+    // the authTagLength is an optional value
+    decryptorBuilder.setAuthTagLength(authTagLength);
+
+    auto authenticatedDecryptor = decryptorBuilder.buildAuthenticatedEncryptor();
+    authenticatedDecryptor->addAssociatedData(authData);
+    authenticatedDecryptor->setAuthTag(aesEncryptResult.authTag);
+
+    try {
+        authenticatedDecryptor->update(aesEncryptResult.ciphertext);
+        authenticatedDecryptor->finish();
+    }  catch (MoCOCrWException &e) {
+        std::cout << "Decryption failed!" << std::endl;
```

Comment:
```suggestion
        std::cout << "Decryption or auth tag verification failed!" << std::endl;
```

I'd extend the message with auth tag failure, just so it's immediately clear how an authentication failure would show up in our API.